policy-packs/aws-advanced/compute.ts

```diff
@@ -13,51 +13,60 @@
 // limitations under the License.
 
 import * as aws from "@pulumi/aws";
-import { Policy, typedRule } from "@pulumi/policy";
-import * as assert from "assert";
+import {
+    ReportViolation,
+    ResourceValidationArgs,
+    ResourceValidationPolicy,
+    validateTypedResource,
+} from "@pulumi/policy";
 
 export function requireApprovedAmisById(
     name: string,
     approvedAmis: string | Iterable<string>,
-): Policy {
+): ResourceValidationPolicy {
     const amis = toStringSet(approvedAmis);
 
     return {
         name: name,
-        description: "Instances should use approved AMIs",
+        description: "Instances should use approved AMIs.",
         enforcementLevel: "mandatory",
-        rules: [
-            typedRule(
-                aws.ec2.Instance.isInstance,
-                it => amis && assert.ok(amis.has(it.ami), "foo"),
-            ),
-            typedRule(
-                aws.ec2.LaunchConfiguration.isInstance,
-                it => amis && assert.ok(amis.has(it.imageId)),
-            ),
-            typedRule(
-                aws.ec2.LaunchTemplate.isInstance,
-                it => amis && assert.ok(it.imageId === undefined || amis.has(it.imageId)),
-            ),
+        validateResource: [
```

Comment:
I know we discussed how it didn't make a lot of sense to have `rules` which accepted a single `rule`, and how the assumption is that most polices will have a single rule, but I found when updating the examples it was just easier to be able to provide multiple validation functions to be called for certain policies that are checking similar things across multiple resource types (like this case). Particularly when using the `validateTypedResource` helper to provide stronger typed resource objects to check. However, instead of making the property name plural, I've left it singular as the majority of policies will only have a single validation function -- but still allow multiple validation functions if it's easier to implement.

More discussion of this in https://github.com/pulumi/pulumi-policy/pull/131